PROVISIONING-METAL.md

```diff
@@ -0,0 +1,169 @@
+optional = true
+```
+
+> **An additional note on network device names**
+> 
+> Bottlerocket’s interface name policies are [specified in this file](https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/release/80-release.link#L6); with name precedence in the following order: onboard, slot, path.
+> What this means is that udev finds and generates the names for the device, and precedence is given in that order.
+> Typically on-board devices are named `eno*`, hot-plug devices are named `ens*`, and if neither of those names are able to be generated, the “path” name is given, i.e `enp*s*f*`.
+
+### Boot Configuration
+
+Bottlerocket uses a neat feature of the Linux kernel called [Boot Configuration](https://www.kernel.org/doc/html/latest/admin-guide/bootconfig.html), which allows a user to pass additional arguments to the kernel command line at runtime.
+The most immediate use of this feature for most users is setting `console` settings so boot messages can be seen on the appropriate consoles.
+
+In order to make use of this feature, one must create an initrd with the desired settings encoded inside it.
+(Bottlerocket technically doesn't use an initrd, so the file is empty save for the encoded boot config data)
+To create the initrd, you must first create a configuration file containing key value pairs for the settings you would like to pass to kernel / init.
+Full syntax is described in the [Boot Config documentation](https://www.kernel.org/doc/html/latest/admin-guide/bootconfig.html#config-file-syntax), but a simple example is provided below that shows the format of console settings as well as an example `systemd` parameter.
+
+The two acceptable prefixes to settings are `kernel` and `init`.
+Settings prefixed with `kernel` are added to the beginning of the kernel command line.
+Settings prefixed with `init` are added to the kernel command line after the `--`, but before any existing init parameters.
+
+In the example below, two console devices are set up, and `systemd`'s log level is set to `debug`.
+It is important to note that the kernel will output to both console devices, but `systemd` outputs to the last device in line (`ttyS1,115200n8` in the example below).
+
+Example Boot Configuration:
+```
+kernel {
+    console = tty0, "ttyS1,115200n8"
+}
+init {
+    systemd.log_level = debug
+}
+```
+
+Bottlerocket provides a command to create the Boot Configuration initrd from your config file, but there are a few prerequisites.
+You will need the [Bottlerocket code repo](https://github.com/bottlerocket-os/bottlerocket), Rust (via [rustup](https://rustup.rs/) or the official site), Docker (20.10.10 or later), and [cargo-make](https://sagiegurari.github.io/cargo-make/) (`cargo install cargo-make`).
+To build the Boot Configuration initrd, create a file called `bootconfig-input` in the root of the Bottlerocket repo containing your desired key/value pair kernel and init arguments.
+Then run:
+```
+cargo make boot-config
+```
```

Comment:
Is there a `docker run` command using the SDK we could provide instead? Otherwise the requirements are quite involved.